server/routes/searchPatients/restrictedPatientSearchFilter.ts

```diff
@@ -0,0 +1,41 @@
+import { PrisonerSearchSummary } from '../../services/prisonerSearchService'
+
+export default class RestrictedPatientSearchFilter {
+  includePrisonerToMove = (prisoner: PrisonerSearchSummary): boolean => {
+    if (this.determinateSentenceAfterCRD(prisoner)) {
+      return false
+    }
+    if (this.recallAfterSED(prisoner)) {
+      return false
+    }
+    return true
+  }
+
+  includePrisonerToAdd = (prisoner: PrisonerSearchSummary): boolean => {
+    if (prisoner.restrictedPatient) {
+      return false
+    }
+    if (!this.releasedToHospital(prisoner)) {
+      return false
+    }
+    if (this.determinateSentenceAfterCRD(prisoner)) {
+      return false
+    }
+    if (this.recallAfterSED(prisoner)) {
+      return false
+    }
+    return true
+  }
+
+  private releasedToHospital = (prisoner: PrisonerSearchSummary): boolean => {
+    return prisoner.lastMovementTypeCode === 'REL' && prisoner.lastMovementReasonCode === 'HP'
+  }
+
+  private determinateSentenceAfterCRD = (prisoner: PrisonerSearchSummary): boolean => {
+    return !prisoner.recall && !prisoner.indeterminateSentence && prisoner.conditionalReleaseDate < new Date()
+  }
+
+  private recallAfterSED = (prisoner: PrisonerSearchSummary): boolean => {
+    return prisoner.recall && prisoner.sentenceExpiryDate < new Date()
```

Comment:
I worried that you should also be excluding `indeterminateSentence` here, but can't find any prisoners on a recall with a `sentenceExpiryDate` set anyway so think we're good